src/sentry/utils/members.py

```diff
@@ -0,0 +1,92 @@
+from django.conf import settings
+from rest_framework import serializers
+
+from sentry import features, roles
+from sentry.models import AuditLogEntryEvent, InviteStatus
+from sentry.signals import member_invited
+from sentry.utils.audit import create_audit_entry_from_user
+
+ERR_CANNOT_INVITE = "Your organization is not allowed to invite members."
+ERR_JOIN_REQUESTS_DISABLED = "Your organization does not allow requests to join."
+
+
+def validate_invitation(member, organization, user_to_approve, allowed_roles):
+    """
+    Validates whether an org has the options to invite members, handle join requests,
+    and that the member role doesn't exceed the allowed roles to invite.
+    """
+    if not features.has("organizations:invite-members", organization, actor=user_to_approve):
+        raise serializers.ValidationError(ERR_CANNOT_INVITE)
+
+    if (
+        organization.get_option("sentry:join_requests") is False
+        and member.invite_status == InviteStatus.REQUESTED_TO_JOIN.value
+    ):
+        raise serializers.ValidationError(ERR_JOIN_REQUESTS_DISABLED)
```

Comment:
Will these functions be used outside of APIs? If that's the case, we might want to raise generic errors here that can be intercepted by the api and other places that are using this function.

Just something simple like:
```
class CannotInvite(Exception):
    # Probably needs a better name
    pass
```
that you can throw.

Reply:
@wedamija are you saying to raise a specific error then have the serializer catch that error and raise a `ValidationError`? I think that makes sense.

Reply:
Yeah exactly, it just makes these functions a little more generic and non-api specific